File: winshoes/Winshoes/Demos/CBuilder/SMTPClient/unitsetup.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "unitsetup.h"
#include "unitmain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TFormSetup *FormSetup;
//---------------------------------------------------------------------------
__fastcall TFormSetup::TFormSetup(TComponent* Owner)
    : TForm(Owner)
{
 ReadRegistry();
}
//---------------------------------------------------------------------------
void __fastcall TFormSetup::ReadRegistry()
{
  TRegistry& regKey = *new TRegistry();
  bool keyGood = regKey.OpenKey("Software\\smtpdemos\\SETUP", false);
  if (keyGood)
  {
    EditServer->Text = regKey.ReadString("Server");
    EditAddress->Text = regKey.ReadString("Email");
  }
  else
  {
    EditServer->Text = "";
    EditAddress->Text = "";
  }
  delete &regKey;
}
//---------------------------------------------------------------------------
void __fastcall TFormSetup::WriteRegistry()
{
  TRegistry& regKey = *new TRegistry();
  regKey.OpenKey("Software\\smtpdemos\\SETUP", true);
  regKey.WriteString("Server", EditServer->Text);
  regKey.WriteString("Email", EditAddress->Text);
  delete &regKey;
}
//---------------------------------------------------------------------------
void __fastcall TFormSetup::FormCreate(TObject *Sender)
{
  ReadRegistry();
  initialEditServer = EditServer->Text;
  initialEditAddress = EditAddress->Text;
}
//---------------------------------------------------------------------------

void __fastcall TFormSetup::ButtonDoneClick(TObject *Sender)
{
// Do not write to Registry unless there is a change of information.
 if (EditServer->Text != initialEditServer ||
      EditAddress->Text != initialEditAddress)
    WriteRegistry();
  Close();
}
//---------------------------------------------------------------------------

 

File: winshoes/Winshoes/Demos/Delphi/SMTP/unitsetup.pas
unit UnitSetup;

interface

uses
  Windows, Messages, IniFiles, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  ComCtrls, ExtCtrls, StdCtrls;

type
  TFormSetup = class(TForm)
    Panel1: TPanel;
    StatusBar: TStatusBar;
    Label1: TLabel;
    Label2: TLabel;
    EditServer: TEdit;
    EditAddress: TEdit;
    ButtonDone: TButton;
    procedure FormCreate(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure ButtonDoneClick(Sender: TObject);
  private
    procedure ReadInifile;
    procedure WriteIniFile;
  public
  end;

var
  FormSetup: TFormSetup;

implementation
{$R *.DFM}

procedure TFormSetup.FormCreate(Sender: TObject);
begin
  ReadIniFile;
end;

procedure TFormSetup.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  WriteIniFile;
end;

procedure TFormSetup.ReadIniFile;
begin
  with TIniFile.Create(ChangeFileExt(Application.EXEName, '.ini')) do try
    EditServer.Text := ReadString('SETUP','Server','');
    EditAddress.Text := ReadString('SETUP','Email','your.address@goes.here');
  finally Free; end;
end;

procedure TFormSetup.WriteIniFile;
begin
  with TIniFile.Create(ChangeFileExt(Application.EXEName, '.ini')) do try
    WriteString('SETUP','Server',EditServer.Text);
    WriteString('SETUP','Email',EditAddress.Text);
  finally Free; end;
end;

procedure TFormSetup.ButtonDoneClick(Sender: TObject);
begin
  Close;
end;

end.
